Android/app/src/main/java/app/intra/net/socks/TLSProbe.java

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2019 Jigsaw Operations LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app.intra.net.socks;
+
+import android.content.Context;
+import android.os.Bundle;
+import app.intra.sys.Names;
+import com.google.firebase.analytics.FirebaseAnalytics;
+import java.io.IOException;
+import java.net.URL;
+import javax.net.ssl.SSLHandshakeException;
+
+// Static utility to check whether the user's connection supports standard TLS sockets.
+class TLSProbe {
+  static final String DEFAULT_URL = "https://www.google.com/";
```

Comment:
I wonder if this URL can introduce false negatives due to being DNS-poisoned, or blocked by IP? Does it even matter, since we don't enalbe the workaround on Result.OTHER_FAILED?

Reply:
Yes, this test is very conservative in that respect, so if the test URL is unreachable, we won't detect that as TLS interception.  My hope is that this will be useful as-is for at least some users, and we can adjust the logic as needed in the future.